lib/registration.pm

```diff
@@ -231,6 +231,12 @@ sub remove_suseconnect_product {
     $version //= scc_version();
     $arch //= get_required_var('ARCH');
     $params //= '';
+    # De-registration fails when module was already de-registered the same day on
+    # the same qcow image (eg if we re-launch the test suite several times...)
+    # so we need to register it again if not registered (if = unless as bash returns 1 = false)
+    if (script_run("SUSEConnect --status-text |grep -a3 python2 |grep Registered")) {
+        script_run("SUSEConnect -p $name/$version/$arch $params");
```

Comment:
this library is widely used, not sure about to change the behavior for all calls...
probably we should do this from outside having a function that give us the status (if doesn't exist in this lib?)
it should be `add_suseconnect_product` so we could still make use of the retries I guess.

Reply:
You're right, this should not be here, this was kinda POC to see what we can do, I should move this elsewhere. I think we will not use zypper_patch and create our own module here.

Reply:
@lemon-suse any suggestion where to move this?

Reply:
What do you think to add condition to position where invoking this function? 
Ex:
remove_suseconnect_product if module registered

If considered it is hard to judge the status of module, just for special ones then can add a setting to control it as a workaround.